src/Stack/Options/SDistParser.hs

```diff
@@ -21,7 +21,8 @@ sdistOptsParser signDefault = SDistOpts <$>
   (long "sig-server" <> metavar "URL" <> showDefault <>
     value "https://sig.commercialhaskell.org" <>
     help "URL") <*>
-  buildPackageOption
+  buildPackageOption <*>
+  strOption (long "tar-files" <> help "If specified, copy all the tar to this dir")
```

Comment:
Unless I'm mistaken, this setup means that the command line options now _require_ a `--tar-files` argument, and we get the old behavior via `--tar-files ""`. Can you instead use `optional (strOption (...))`?

Also, I think `--tar-path` may be a nicer option name.

Reply:
But if I make it optional and `sdoptsTarPath` will become `Maybe FilePath` I get this error, that 
```
2 of 2] Compiling Main             ( src/main/Main.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-1.24.2.0/build/stack/stack-tmp/Main.o ) [Stack.Options.SDistParser changed]

stack/src/main/Main.hs:762:40: error:
    • Couldn't match type ‘Maybe FilePath’ with ‘[Char]’
      Expected type: FilePath
        Actual type: Maybe FilePath
    • In the second argument of ‘($)’, namely ‘sdoptsTarPath sdistOpts’
      In a stmt of a 'do' block:
        targetDir <- resolveDir' $ sdoptsTarPath sdistOpts
      In the second argument of ‘($)’, namely
        ‘do { targetDir <- resolveDir' $ sdoptsTarPath sdistOpts;
              targetTarPath <- (targetDir </>) <$> parseRelFile tarName;
              D.createDirectoryIfMissing True (toFilePath targetDir);
              D.copyFile (toFilePath tarPath) (toFilePath targetTarPath) }’
```
As far as I understand, it says that `resolveDir'` can accept only `String` but not `Maybe String`, what should I do in such a case?

Reply:
You're probably going to need to pattern match in that part of the code to determine whether to use the original default path or this new configuration value.